test/intl402/Temporal/Calendar/prototype/yearMonthFromFields/reference-day.js

```diff
@@ -0,0 +1,133 @@
+  result4,
+  5782, 4, "M04",
+  "reference day is the first of the calendar month even if day is given",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 5
+);
+const isoFields = result4.getISOFields();
+assert.sameValue(isoFields.isoYear, 2021, "Tevet 5782 begins in ISO year 2021");
+assert.sameValue(isoFields.isoMonth, 12, "Tevet 5782 begins in ISO month 12");
+
+const result5 = hebrew.yearMonthFromFields({ year: 5783, monthCode: "M05L" }, { overflow: "constrain" });
+TemporalHelpers.assertPlainYearMonth(
+  result5,
+  5783, 5, "M05",
+  "month code M05L does not exist in year 5783 (overflow constrain)",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 23
+);
+
+assert.throws(
+  RangeError,
+  () => hebrew.yearMonthFromFields({ year: 5783, monthCode: "M05L" }, { overflow: "reject" }),
+  "month code M05L does not exist in year 5783 (overflow reject)",
+);
+
+const result6 = hebrew.yearMonthFromFields({ year: 5783, month: 13 }, { overflow: "constrain" });
+TemporalHelpers.assertPlainYearMonth(
+  result6,
+  5783, 12, "M12",
+  "month 13 does not exist in year 5783 (overflow constrain)",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 18
+);
+
+assert.throws(
+  RangeError,
+  () => hebrew.yearMonthFromFields({ year: 5783, month: 13 }, { overflow: "reject" }),
+  "month 13 does not exist in year 5783 (overflow reject)",
+);
+
+const result7 = hebrew.yearMonthFromFields({ year: 5782, monthCode: "M04", day: 50 }, { overflow: "constrain" });
+TemporalHelpers.assertPlainYearMonth(
+  result7,
+  5782, 4, "M04",
+  "reference day is set correctly even if day is out of range (overflow constrain)",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 5
+);
+
+const result8 = hebrew.yearMonthFromFields({ year: 5782, monthCode: "M04", day: 50 }, { overflow: "reject" });
+TemporalHelpers.assertPlainYearMonth(
+  result8,
+  5782, 4, "M04",
+  "reference day is set correctly even if day is out of range (overflow reject)",
+  /* era = */ undefined, /* era year = */ undefined, /* reference day = */ 5
+);
+
+const chinese = new Temporal.Calendar("chinese");
+
+// Month codes, month indices, and the ISO reference days of the months in 2022
+const testData = [
+  // TODO: Sources conflict over whether M01L and M12L exist in _any_ year.
+  // Clarify this, and uncomment or delete as appropriate. ICU says no, and most
+  // implementations are backed by ICU, so for now we won't test for it.
+  // ["M01", 1, 1],
+  ["M02", 2, 3],
+  ["M03", 3, 1],
+  ["M04", 4, 1],
+  ["M05", 5, 30],
+  ["M06", 6, 29],
+  ["M07", 7, 29],
+  ["M08", 8, 27],
+  ["M09", 9, 26],
+  ["M10", 10, 25],
+  ["M11", 11, 24],
+  // ["M12", 12, 23],
+];
+for (const [monthCode, month, referenceISODay] of testData) {
+  const fields = { year: 2022, monthCode: monthCode + "L" };
+
+  const result = chinese.yearMonthFromFields(fields, { overflow: "constrain" });
```

Comment:
Should you also test years where those leap months *should* exist?  Here's some code you can use to find appropriate dates to use: 

```js
lines = [];
formatter = new Intl.DateTimeFormat('en', { calendar: 'chinese' });
for (year = 100; year < 3000; year++) {
  for (month = 1; month <= 12; month++) {
    const date = new Date(year, month-1, 1);
    const parts = formatter.formatToParts(date);
    const monthString = parts.filter(p => p.type === 'month')[0].value;
    const isLeapMonth = monthString.endsWith('bis');
    if (isLeapMonth) lines.push( `${date.toISOString()}\t${monthString}\t${formatter.format(date)}`);
  }
}
console.log(lines.join('\n'));
```

Reply:
Thanks for that script, it was really useful. I'm adding a note that these leap months are according to ICU, since I don't otherwise have any sources.